### PR TITLE
5 guest creates account

### LINKED ISSUE #5: Guest Can Create New Account
As a guest,
When I visit "/",
And I click on "Create Account",
I am redirected to "/sign_up",
And I see fields for:
- First Name
- Last Name
- Username
- Email
- Street Address
- City
- State
- Zipcode
- Bio
- Picture (PaperClip)
- Password
- Password Confirmation
- Role

And when I fill in all the fields,
And I select "Lister" for my role,
And then I click on "Create User",
Then I am redirected to "/dashboard"
And on the dashboard I see my new information,
And I see the "Create Job" button.

### PR DESCRIPTION
**Pull request by danjwinter**
Add fields for bio and picture upload.
Make dashboard dynamic to display user info.
Make dashboard dynamic to display links and tables based on user role.
closes #5

### CODE REVIEW

File: app/views/user/users/_form.html.erb [hunk 1]
```diff
@@ -37,12 +37,21 @@
...
+
   <% unless current_user %>
     <%= f.label :username %>
     <%= f.text_field :username  %>
   <% end %>
   <br>
 
+  <%= f.label :file_upload %>
+  <br><br>
+  <%= f.file_field :file_upload, class: "upload_file grey-text" %>
```
Comment (ToniRib) [inline app/views/user/users/_form.html.erb:52]: I think when we cleaned up the css we either removed the upload_file class or renamed it to upload-file

File: app/views/user/users/_lister_dashboard_links.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    </div>
+  </div>
+  <div class="col s12 m4 l4">
+    <div class="user-action-links">
+      <%= link_to "Edit Profile", "#" %>
+    </div>
+  </div>
+  <div class="col s12 m4 l4">
+    <div class="user-action-links">
+      <%= link_to "Public Profile", "/users/dan_user" %>
```
Comment (ToniRib) [inline app/views/user/users/_lister_dashboard_links.html.erb]: Can we make this an actual link (user_path(@user)) so I can use it in my branch?

**Comment by ToniRib** [comment]
Don't forget to merge in the latest master before we pull this in.

File: app/views/user/users/_form.html.erb (hunk 1 above)
Comment (danjwinter) [inline app/views/user/users/_form.html.erb:52]: Correct, class removed.

File: app/views/user/users/_lister_dashboard_links.html.erb (hunk 2 above)
Comment (danjwinter) [inline app/views/user/users/_lister_dashboard_links.html.erb]: Added link to @user

**Comment by ToniRib** [comment]
Looks good, thanks for fixing those things @danjwinter. @jwperry do you want to merge this in?

**Comment by jwperry** [comment]
Looks good, solid job with link tie-in.

**Pull request merged**